data-prepper-plugins/otel-proto-common/src/main/java/com/amazon/dataprepper/plugins/otel/codec/OTelProtoCodec.java

```diff
@@ -457,4 +486,291 @@ protected AnyValue objectToAnyValue(final Object obj) throws UnsupportedEncoding
+     *
+     * @return A map, containing information about the instrumentation library
+     */
+    public static Map<String, Object> getInstrumentationLibraryAttributes(final InstrumentationLibrary instrumentationLibrary) {
+        final Map<String, Object> instrumentationAttr = new HashMap<>();
+        if (!instrumentationLibrary.getName().isEmpty()) {
+            instrumentationAttr.put(INSTRUMENTATION_LIBRARY_NAME, instrumentationLibrary.getName());
+        }
+        if (!instrumentationLibrary.getVersion().isEmpty()) {
+            instrumentationAttr.put(INSTRUMENTATION_LIBRARY_VERSION, instrumentationLibrary.getVersion());
+        }
+        return instrumentationAttr;
+    }
+
+    /**
+     * Extracts the name and version of the used instrumentation scope used
+     *
+     * @return A map, containing information about the instrumentation scope
+     */
+    public static Map<String, Object> getInstrumentationScopeAttributes(final InstrumentationScope instrumentationScope) {
+        final Map<String, Object> instrumentationScopeAttr = new HashMap<>();
+        if (!instrumentationScope.getName().isEmpty()) {
+            instrumentationScopeAttr.put(INSTRUMENTATION_SCOPE_NAME, instrumentationScope.getName());
+        }
+        if (!instrumentationScope.getVersion().isEmpty()) {
+            instrumentationScopeAttr.put(INSTRUMENTATION_SCOPE_VERSION, instrumentationScope.getVersion());
+        }
+        return instrumentationScopeAttr;
+    }
+
+
+    public static String getStartTimeISO8601(final NumberDataPoint numberDataPoint) {
+        return convertUnixNanosToISO8601(numberDataPoint.getStartTimeUnixNano());
+    }
+
+    public static String getTimeISO8601(final NumberDataPoint ndp) {
+        return convertUnixNanosToISO8601(ndp.getTimeUnixNano());
+    }
+
+    public static Optional<String> getServiceName(final Resource resource) {
+        return resource.getAttributesList().stream()
+                .filter(keyValue -> keyValue.getKey().equals(SERVICE_NAME) && !keyValue.getValue().getStringValue().isEmpty())
+                .findFirst()
+                .map(i -> i.getValue().getStringValue());
+    }
+
+
+    public static Map<String, Object> mergeAllAttributes(final Collection<Map<String, Object>> attributes) {
+        return attributes.stream()
+                .flatMap(map -> map.entrySet().stream())
+                .collect(Collectors.toMap(Map.Entry::getKey, Map.Entry::getValue));
+    }
+
+
+    public static List<Quantile> getQuantileValues(List<SummaryDataPoint.ValueAtQuantile> quantileValues) {
+        return quantileValues.stream()
+                .map(q -> new DefaultQuantile(q.getQuantile(), q.getValue()))
+                .collect(Collectors.toList());
+    }
+
+    /**
+     * Create the buckets, see <a href="https://github.com/open-telemetry/opentelemetry-proto/blob/main/opentelemetry/proto/metrics/v1/metrics.proto">
+     *     the OTel metrics proto spec</a>
+     * <p>
```

Comment:
The Gradle build is failing because these Javadocs are invalid. Please see #1271 for a possible solution.

https://github.com/opensearch-project/data-prepper/runs/6182249051?check_suite_focus=true